File: project_euler/ruby/euler_6.rb
# Problem #6: Sum square difference
def sum_square_difference
  sum_square = 0
  square_sum = 0

  (1..100).each do |x|
    sum_square += x**2
    square_sum += x
  end

  square_sum **= 2

  puts "The difference is #{square_sum - sum_square}" # Output: 25164150
end

File: project_euler/js/euler_6.js
// Problem #6: Sum square difference
function sumSquareDifference() {
  var sumSquare = 0;
  var squareSum = 0;

  for (var i = 1; i <= 100; i++) {
    sumSquare += Math.pow(i, 2);
  }

  for (var j = 1; j <= 100; j++) {
    squareSum += j;
  }
  squareSum = Math.pow(squareSum, 2);
  return squareSum - sumSquare; // Output: 25164150
}
